sympy/sets/tests/test_sets.py

```diff
@@ -1597,6 +1597,27 @@ def test_DisjointUnion_len():
     assert len(DisjointUnion(S.EmptySet, S.EmptySet, FiniteSet(x, y, z), S.EmptySet)) == 3
     raises(ValueError, lambda: len(DisjointUnion(Interval(0, 1), S.EmptySet)))
 
+def test_SetKind_ProductSet():
+    assert isinstance(ProductSet(
+        FiniteSet(Matrix([1, 2])), FiniteSet(Matrix([1, 2]))).kind, SetKind)
+    assert isinstance(ProductSet(
+        Interval(1, 2), FiniteSet(Matrix([1, 2]))).kind, SetKind)
+
+def test_SetKind_Interval():
+    assert isinstance(Interval(1, 2).kind, SetKind)
+
+def test_SetKind_EmptySet_UniversalSet():
+    assert isinstance(EmptySet.kind, SetKind)
+    assert isinstance(UniversalSet.kind, SetKind)
```

Comment:
Only asserting `isinstance` misses most of what the test should be testing.

Reply:
@oscarbenjamin